### ISSUE TITLE
enhanced whitelist

### ISSUE THREAD

**Issue by cadeyrn** [cadeyrn/bookmarks-organizer#96]
A whitelist has been added in Bookmarks Organizer 3.0. This includes a link to add bookmarks to the whitelist and the ability to reset the whitelist.

In a future update there will be an enhanced whitelist which lets you remove single entries from the whitelist.

**Comment by DuMuT6p** [comment]
Hello, 
Is there a way currently to view the whitelist ?

**Comment by cadeyrn** [comment]
No. This issue is about implementing the list view and the option to remove entries from the whitelist. You can only clear the whitelist in the add-on's settings in the current version.

**Comment by DuMuT6p** [comment]
Thank you. This seems like a great feature to have.

**Comment by Illegal-Services** [comment]
Hello, I've read the source code from the whitelist and tried to apply a merge request with export/import buttons, but I have noticed that you uses of bookmark nodes in order to whitelist.
In my opinion this is really wrong because if we change profile and whitelist the exact same bookmark, it will be another node.
Can you consider re-writting this feature, and makes uses of the literal string url instead of the node?

**Comment by Illegal-Services** [comment]
Making this change will be the first step to resolve this issues:
https://github.com/cadeyrn/bookmarks-organizer/issues/96:   enhanced whitelist
https://github.com/cadeyrn/bookmarks-organizer/issues/115:  [request] whitelisting webpage domains before scan
https://github.com/cadeyrn/bookmarks-organizer/issues/182: [FEATURE REQUEST/QUESTION] Whitelisting source or destination address?
https://github.com/cadeyrn/bookmarks-organizer/issues/202: [FEATURE REQUEST] - Add an option to export and import whitelist (settings)

**Comment by Illegal-Services** [comment]
Personally on my own project, that is in Python, I whitelist using:

[{HTTP_Code}]: {Source URL} > [{HTTP_Code}]: {Destination URL}
[301]: https://www.google.com/ > [200]: https://www.youtube.com/

(As many redirects as there are 👆)

And for "error" bookmarks 
I use:
{HTTP_code} {URL}
[404] https://www.google.com

Exactly has you mentioned, it contains downsides as well.

🤷

**Comment by Illegal-Services** [comment]
I also make uses of regex in my whitelist, which is useful for certain websites who their URL changes every x time.
